database/misc/redis.py

```diff
@@ -257,15 +253,12 @@ def main():
 
     # flush Command section -----------
     elif command == "flush":
-        try:
-            db = int(module.params['db'])
-        except Exception:
-            db = 0
+        db = module.params['db']
         mode = module.params['flush_mode']
 
         #Check if we have all the data
         if mode == "db":
-            if type(db) != int:
+            if db is None:
                 module.fail_json(
```

Comment:
I do not know redis enough to know what should be done. I think we should wait for the maintainer of the module to tell what he wanted to express with that construct, but yeah, it would be more logical to default to 0.

But "anything not a number as a argument mean the first db" is a bit curious in the first place.